src/components/income/OtherInvestments.tsx

```diff
@@ -0,0 +1,198 @@
+  }
+})
+
+interface AssetUserInput {
+  name: string
+  positionType: AssetType
+  openDate?: Date
+  closeDate?: Date
+  openPrice: string
+  closePrice: string
+  quantity: string
+  state?: State
+}
+
+const blankUserInput: AssetUserInput = {
+  name: '',
+  positionType: 'Security',
+  openPrice: '',
+  closePrice: '',
+  quantity: ''
+}
+
+const toUserInput = (f: Asset<Date>): AssetUserInput => ({
+  ...blankUserInput,
+  ...f,
+  openPrice: f.openPrice.toString(),
+  closePrice: f.closePrice?.toString() ?? '',
+  quantity: f.quantity.toString()
+})
+
+const toAsset = (input: AssetUserInput): Asset<Date> | undefined => {
+  const { name, openDate, closeDate, openPrice, closePrice, quantity, state } =
+    input
+  if (name === '' || openDate === undefined) {
+    return undefined
+  }
+  return {
+    name,
+    openDate,
+    closeDate,
+    openPrice: Number(openPrice),
+    closePrice: Number(closePrice),
+    quantity: Number(quantity),
+    state
+  }
+}
+
+export default function OtherInvestments(): ReactElement {
+  const assets = useSelector((state: YearsTaxesState) => state.assets)
+
+  const methods = useForm<AssetUserInput>()
+  const { handleSubmit, watch } = methods
+  const positionType = watch('positionType')
+  const dispatch = useDispatch()
+
+  const { onAdvance, navButtons } = usePager()
+
+  const onSubmitAdd = (formData: AssetUserInput): void => {
+    const payload = toAsset(formData)
+    if (payload !== undefined) {
+      dispatch(addAsset(payload))
+    }
+  }
+
+  const onSubmitEdit =
+    (index: number) =>
+    (formData: AssetUserInput): void => {
+      const payload = toAsset(formData)
+      if (payload !== undefined) {
+        dispatch(editAsset({ value: payload, index }))
+      }
+    }
+
+  const form: ReactElement | undefined = (
+    <FormListContainer<AssetUserInput>
+      onSubmitAdd={onSubmitAdd}
+      onSubmitEdit={onSubmitEdit}
+      items={assets.map((a) => toUserInput(a))}
+      removeItem={(i) => dispatch(removeAsset(i))}
+      primary={(f) => f.name}
+      secondary={(f) =>
+        `${f.openDate?.toDateString() ?? ''}: ${f.quantity} @ $${f.openPrice}`
+      }
+    >
+      {' '}
+      <Grid container spacing={2}>
+        <p>Manage Assets</p>
+        <GenericLabeledDropdown<AssetType>
+          label="Asset Type"
+          name="positionType"
+          dropDownData={['Security', 'Real Estate']}
+          keyMapping={(x) => x}
+          textMapping={show(showAssetType)}
+          valueMapping={(x) => x}
+        />
+        <LabeledInput
+          label="Address"
+          patternConfig={Patterns.name}
+          name="name"
+        />
+        <DatePicker
+          maxDate={new Date(2021, 12, 31)}
+          label={positionType === 'Real Estate' ? 'Purchase date' : 'Open Date'}
+          name="openDate"
+        />
+        <DatePicker
+          maxDate={new Date(2021, 12, 31)}
+          label={positionType === 'Real Estate' ? 'Sale date' : 'Close Date'}
```

Comment:
Just as an average person I think "purchase date" (or maybe acquisition date instead) and "sale date" make more sense to me for both Real Estate and Securities than "open/close" dates

Reply:
You're right. There are some cases where selling happens before buying, but I'll stick to the wording on the 8949 itself, "date acquired" and "date sold or disposed of".